accession-commons-monotonic-generator-jpa/src/test/java/uk/ac/ebi/ampt2d/commons/accession/core/BasicMonotonicAccessioningWithAlternateRangesTest.java

```diff
@@ -65,49 +66,53 @@ public void testAlternateRangesWithDifferentGenerators() throws AccessionCouldNo
         String categoryId = "eva_2";
         String instanceId2 = "test-instance_2";
         List<AccessionWrapper<TestModel, String, Long>> evaAccessions = getAccessioningService(categoryId, INSTANCE_ID)
-                .getOrCreate(getObjectsForAccessionsInRange(1, 10));
+                .getOrCreate(getObjectsForAccessionsInRange(1, 9));
         assertEquals(9, evaAccessions.size());
-        assertEquals(1, evaAccessions.get(0).getAccession().longValue());
-        assertEquals(9, evaAccessions.get(8).getAccession().longValue());
+        assertEquals(0, evaAccessions.get(0).getAccession().longValue());
+        assertEquals(8, evaAccessions.get(8).getAccession().longValue());
         assertEquals(1, contiguousIdBlockService
                 .getUncompletedBlocksByCategoryIdAndApplicationInstanceIdOrderByEndAsc(categoryId, INSTANCE_ID)
                 .size());
 
         //get another service for same category
         evaAccessions = getAccessioningService(categoryId, INSTANCE_ID)
-                .getOrCreate(getObjectsForAccessionsInRange(10, 30));
+                .getOrCreate(getObjectsForAccessionsInRange(11, 30));
         assertEquals(20, evaAccessions.size());
         //Previous block ended here
-        assertEquals(10, evaAccessions.get(0).getAccession().longValue());
-        //New Block after 200 values interval
-        assertEquals(211, evaAccessions.get(1).getAccession().longValue());
-        assertEquals(220, evaAccessions.get(10).getAccession().longValue());
-        assertEquals(421, evaAccessions.get(11).getAccession().longValue());
-        assertEquals(429, evaAccessions.get(19).getAccession().longValue());
+        assertEquals(9, evaAccessions.get(0).getAccession().longValue());
+
+        //New Block still not interleaved
+        assertEquals(10, evaAccessions.get(1).getAccession().longValue());
+        assertEquals(19, evaAccessions.get(10).getAccession().longValue());
+
+        //New Block interleaved
+        assertEquals(40, evaAccessions.get(11).getAccession().longValue());
+        assertEquals(48, evaAccessions.get(19).getAccession().longValue());
```

Comment:
If the beginning of the block below is 50, shouldn't this test against 49?

Reply:
nope 49 is not accessioned still , it is accessioned again below  //get previous uncompleted service from instance1 and create accessions